torch/csrc/jit/mobile/model_compatibility.cpp

```diff
@@ -0,0 +1,91 @@
+#include <ATen/core/ivalue.h>
+#include <caffe2/serialize/file_adapter.h>
+#include <caffe2/serialize/inline_container.h>
+#include <torch/csrc/jit/mobile/model_compatibility.h>
+#include <torch/csrc/jit/serialization/import_read.h>
+
+#include <string>
+#include <vector>
+
+namespace torch {
+namespace jit {
+
+using caffe2::serialize::FileAdapter;
+using caffe2::serialize::IStreamAdapter;
+using caffe2::serialize::PyTorchStreamReader;
+using caffe2::serialize::ReadAdapterInterface;
+
+namespace {
+
+c10::IValue readArchive(
+    const std::string& archive_name,
+    PyTorchStreamReader& stream_reader) {
+  c10::optional<at::Device> device;
+
+  auto ivalues = torch::jit::readArchiveAndTensors(
+      archive_name, nullptr, nullptr, device, stream_reader);
```

Comment:
I was referring to this. Does this work only in some cases?

Reply:
Yup it only works for in some minimum cases.